### PR TITLE
`stackcheck.h`

### PR DESCRIPTION
**Pull request by AngheloAlf**
Just a simple header and cleanup for stackcheck

### CODE REVIEW

File: src/boot_O2/stackcheck.c [hunk 1]
```diff
@@ -32,27 +33,28 @@ void StackCheck_Init(StackEntry* entry, void* stackTop, void* stackBottom, u32 i
         }
 
         sStackInfoListEnd = entry;
-        if (!sStackInfoListStart) {
+        if (sStackInfoListStart == 0) {
```
Comment (engineer124) [inline src/boot_O2/stackcheck.c:36]:
```suggestion
        if (sStackInfoListStart == NULL) {
```

File: src/boot_O2/stackcheck.c [hunk 2]
```diff
@@ -32,27 +33,28 @@ void StackCheck_Init(StackEntry* entry, void* stackTop, void* stackBottom, u32 i
...
             }
         }
     }
 }
 
 void StackCheck_Cleanup(StackEntry* entry) {
-    u32 inconsistency = 0; // unused variable needed to match
+    u32 inconsistency = false;
 
     if (!entry->prev) {
```
Comment (engineer124) [inline src/boot_O2/stackcheck.c:53]:
```suggestion
    if (entry->prev == NULL) {
```

File: src/boot_O2/stackcheck.c [hunk 3]
```diff
@@ -71,22 +73,22 @@ void StackCheck_Cleanup(StackEntry* entry) {
...
 
-    used = entry->tail - (u32)last;
-    free = (u32)last - entry->head;
+    used = (uintptr_t)entry->tail - (uintptr_t)last;
+    free = (uintptr_t)last - (uintptr_t)entry->head;
 
     if (free == 0) {
         status = STACK_STATUS_OVERFLOW;
-    } else if (free < (u32)entry->minSpace && entry->minSpace != -1) {
+    } else if (free < (size_t)entry->minSpace && entry->minSpace != -1) {
```
Comment (engineer124) [inline src/boot_O2/stackcheck.c:91]:
```suggestion
    } else if ((free < (size_t)entry->minSpace) && (entry->minSpace != -1)) {
```

File: src/boot_O2_g3/boot_main.c [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #include "prevent_bss_reordering.h"
 #include "global.h"
 #include "stack.h"
+#include "stackcheck.h"
```
Comment (hensldm) [inline src/boot_O2_g3/boot_main.c:4]: Just below here could you remove the word 'Thread' from the two stack variables. None of the other stack variables have it.

**Review by hensldm — APPROVED** [review]

File: include/stackcheck.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    /* 0x04 */ struct StackEntry* prev;
+    /* 0x08 */ void* head;
+    /* 0x0C */ void* tail;
+    /* 0x10 */ u32 initValue;
+    /* 0x14 */ s32 minSpace;
+    /* 0x18 */ const char* name;
+} StackEntry; // size = 0x1C
+
+
+void StackCheck_Init(StackEntry* entry, void* stackTop, void* stackBottom, u32 initValue, s32 minSpace, const char* name);
```
Comment (hensldm) [inline include/stackcheck.h:24]: Are `stackTop` and `stackBottom` reversed? The third arg seems to use the `STACK_TOP` macro.

File: include/stackcheck.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#ifndef STACKCHECK_H
+#define STACKCHECK_H
+
+#include "ultra64.h"
+
+
```
Comment (tom-overton) [inline include/stackcheck.h:6]: Looks like there's an extra newline here.

File: include/stackcheck.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    /* 0x00 */ struct StackEntry* next;
+    /* 0x04 */ struct StackEntry* prev;
+    /* 0x08 */ void* head;
+    /* 0x0C */ void* tail;
+    /* 0x10 */ u32 initValue;
+    /* 0x14 */ s32 minSpace;
+    /* 0x18 */ const char* name;
+} StackEntry; // size = 0x1C
+
+
```
Comment (tom-overton) [inline include/stackcheck.h:23]: Same thing here.

**Review by tom-overton — APPROVED** [review]
Just the most minor of newline gripes, but I'll still approve.

**Pull request merged**